homeassistant/components/tts/entity.py

```diff
@@ -0,0 +1,159 @@
+"""Entity for Text-to-Speech."""
+
+from collections.abc import Mapping
+from functools import partial
+from typing import Any, final
+
+from propcache.api import cached_property
+
+from homeassistant.components.media_player import (
+    ATTR_MEDIA_ANNOUNCE,
+    ATTR_MEDIA_CONTENT_ID,
+    ATTR_MEDIA_CONTENT_TYPE,
+    DOMAIN as DOMAIN_MP,
+    SERVICE_PLAY_MEDIA,
+    MediaType,
+)
+from homeassistant.const import ATTR_ENTITY_ID, STATE_UNAVAILABLE, STATE_UNKNOWN
+from homeassistant.core import callback
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.util import dt as dt_util
+
+from .const import TtsAudioType
+from .media_source import generate_media_source_id
+from .models import Voice
+
+CACHED_PROPERTIES_WITH_ATTR_ = {
+    "default_language",
+    "default_options",
+    "supported_languages",
+    "supported_options",
+}
+
+
+class TextToSpeechEntity(RestoreEntity, cached_properties=CACHED_PROPERTIES_WITH_ATTR_):
```

Comment:
Please do a straight 1:1 move of the entity class to this module in a separate PR first so it's easier to review this PR.